test/language/statements/try/optional-catch-binding-throws.js

```diff
@@ -0,0 +1,25 @@
+// Copyright (C) 2017 Lucas Azzola. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+author: Lucas Azzola
+esid: pending
+description: errors can be thrown from catch clause without binding
+features: [optional-catch-binding]
+info: |
+  Runtime Semantics: CatchClauseEvaluation
+
+  Catch : catch Block
+    (...)
+    Let B be the result of evaluating Block.
+    (...)
+    Return Completion(B).
+---*/
+
+assert.throws(Test262Error, function() {
```

Comment:
@michaelficarra, The description says it is checking the `catch` block won't hold new abrupt completions.

---

Anyway your code example it might be even more simple, but this is already being very well tested at test/language/statements/try/optional-catch-binding-lexical.js

```js
let x = false;

try {
  throw 0;
} catch {
  x = true
}

assert(x);
```